File: graph_input_problems/degree_euler_graph_step1/swift/step1.swift
/*
これはpaizaラーニングの「グラフ構造の入力メニュー」-「頂点の出現回数・無向グラフ」に
swiftでチャレンジしたコードです。
https://paiza.jp/works/mondai/graph_input_problems/graph_input_problems__degree_euler_graph_step1

作成環境
Ubuntu 20.04.2 LTS
Swift version 5.4 (swift-5.4-RELEASE) Target: x86_64-unknown-linux-gnu
*/
import Foundation

// メイン関数
func main() {
    // 1 行目に、頂点の個数を表す整数 n と、頂点の組の個数を表す整数 m, 頂点 v が半角スペース区切りで与えられます。
    let items = readLine()!.trimmingCharacters(in: .whitespacesAndNewlines).split(separator: " ")
    //let n = Int(items[0])!
    let m = Int(items[1])!
    let v = Int(items[2])!
    // 続く m 行では、頂点の組 a_i, b_i が半角スペース区切りで与えられます。(1 ≦ i ≦ m)
    var answer: Int = 0
    for _ in 0..<m {
        let itemsG = readLine()!.trimmingCharacters(in: .whitespacesAndNewlines).split(separator: " ")
        let a = Int(itemsG[0])!
        let b = Int(itemsG[1])!
        if a == v || b == v {
            // 頂点につながる辺をカウント
            answer += 1
        }
    }
    // 結果出力
    print(answer)
}

// エントリーポイント
main()

File: graph_input_problems/degree_euler_graph_step1/golang/step1.go
/*
これはpaizaラーニングの「グラフ構造の入力メニュー」-「頂点の出現回数・無向グラフ」に
go言語でチャレンジしたコードです。
https://paiza.jp/works/mondai/graph_input_problems/graph_input_problems__degree_euler_graph_step1

作成環境
Ubuntu 20.04.2 LTS
go version go1.16.4 linux/amd64
*/

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// main はエントリーポイント
func main() {
	var sc *bufio.Scanner = bufio.NewScanner(os.Stdin)
	// 1 行目に、頂点の個数を表す整数 n と、頂点の組の個数を表す整数 m, 頂点 v が半角スペース区切りで与えられます。
	sc.Scan()
	var items []string = strings.Split(strings.TrimSpace(sc.Text()), " ")
	//var n, _ = strconv.Atoi(items[0])
	var m, _ = strconv.Atoi(items[1])
	var v, _ = strconv.Atoi(items[2])
	// 続く m 行では、頂点の組 a_i, b_i が半角スペース区切りで与えられます。(1 ≦ i ≦ m)
	var answer int = 0
	for i := 0; i < m; i++ {
		sc.Scan()
		var items []string = strings.Split(strings.TrimSpace(sc.Text()), " ")
		var a, _ = strconv.Atoi(items[0])
		var b, _ = strconv.Atoi(items[1])
		if a == v || b == v {
			// 頂点につながる辺をカウント
			answer++
		}
	}
	// 答えを出力する
	fmt.Println(answer)
	return
}
